### ISSUE TITLE
implementation of mixture of experts by keras

### ISSUE THREAD

**Issue by mmoradik** [fchollet/keras#3370]
Hi, I am trying to implement mixture of expert neural networks using keras. 

For instance, I have 3 expert neural net, each have 2 outputs. A gated network assigns a probability to each expert. The gated network in my case has 3 output generated by softmax. Inputs to all experts and gated network are same.

My question is how I can customize the loss function in the following form:
loss=sum_i(p_i*(y_true-y_i)^2)
where p_i is the probability that gated network assigns to expert i, y_true is the desired output (in my case 2 dimensional), and y_i is the output of expert I. and sum is on the number of experts.

merged = Merge([gated, expert1, expert2,expert3], mode='concat')
model = Sequential()
model.add(merged)
model.compile(loss=loss function, optimizer=Rmsprop)
model.fit([in,in,in,in],[out,out,out])

**Comment by MaryamAkhavan** [comment]
Hi, 
Could you impelemnt your code with loss function?
I faced with this problem. If you did, can you help me?
thank you

**Comment by MaryamAkhavan** [comment]
Hi, 
Could you impelemnt your code with loss function?
I faced with this problem. If you did, can you help me?
thank you
